### ISSUE TITLE
Tumbleweed install is inaccessible over SSH

### ISSUE THREAD

**Issue by RlndVt** [rockstor/rockstor-core#2765]
With the current configuration: SSH is only accessible by root, and also only by not using a password.

It is further not possible to add a public key for authentication.

Suggestions:
- (Add option to) open SSH to non root users, possibly only to members of a `ssh` group
- Add the possibility to add public keys to UI
- Open root authentication by password. (Not my preference, see PPS.)

PS. See #2277 for additional background.

PPS. Slight tangent, but I would additionally suggest a warning against having root SSH accessible by password. Leaving this open places immense trust in the local network, and or the chosen root password; which I would strongly dissuade.

**Comment by FroggyFlox** [comment]
Hi @RlndVt, thanks a lot for creating this issue!

Thanks also for the feedback on possible improvements as well and listing your suggestions accordingly. There is a lot to unpack here and each of these points possibly deserves its own issue and discussion. This indeed touches a very important part of the user experience because we don't want to create a situation where the user would suddenly not be able to remotely access their Rockstor machine as they used to. We thus need to carefully address these. The most pressing point to be addressed here is to restore SSH access as we currently do to our Tumbleweed offering so I'll take the opportunity to create a dedicated issue. We really need to be as focused as possible on each issue to help with getting it done faster, more easily, and notably be able to cherry-pick/rollback or simply better isolate each modification to the code base. This is thus the focus of https://github.com/rockstor/rockstor-core/issues/2767.

I've thus also renamed this very issue to better represent its intent: adopting upstream defaults with regards to SSH access. Sorry for that but I'm sure you understand; I hope you don't mind. This would, however, require quite a bit of work as a few things would need to ensured/in order before we can "turn that ON". You've started laying these out already, but let's expand a bit what I can think of for now:

- Ensure compatibility between both our OS bases: Leap and Tumbleweed. If both have different defaults (which seems to be the case for now but needs confirmation), this does increase the complexity quite a bit.
- We need to make sure none of that would break existing SFTP configurations, as this is the "other" main use of openSSH in Rockstor.
- Similarly, we would need to ensure a very robust "migration" mechanism so that we don't suddenly break users' access on existing installations. As a result, **this would first require a very good coverage by our unit tests**.
- Add SSH key upload facility. This would also require the implementation of an "interaction layer" between Rockstor and openSSH.


The biggest question, and probably most difficult to implement: should this be the default, or should we provide all the tools to move to the respective upstream's defaults in a very user-friendly way? I personally would tend to lean towards the latter as that would be the most user-friendly way, while providing the user with all they need to easily lock things down as needed. This would create a more obvious realization of the actual consequences to the less knowledgeable user. It is a very difficult decision, though, and I'm not even talking about the work that would be needed for that (more detailed above).  Food for thought on my end for now...

**Comment by Hooverdan96** [comment]
Some additional info on the repackaging on `/usr/` vs. `/etc/`:

https://en.opensuse.org/openSUSE:Packaging_UsrEtc
